api/v01/vrp.rb

```diff
@@ -36,573 +37,114 @@ def self.call(object, _env)
-            end
-
-            exactly_one_of :vrp, :vehicles # either it is a json (and :vrp is required) or it is a csv (and :vehicles is required)
+            use(:input)
           }
-          post do
-            begin
-              # Api key is not declared as part of the VRP and must be handled carefully and separatly from other parameters
-              api_key = params[:api_key]
-              checksum = Digest::MD5.hexdigest Marshal.dump(params)
-              d_params = declared(params, include_missing: false)
-              vrp_params = d_params[:points] ? d_params : d_params[:vrp]
-              APIBase.dump_vrp_dir.write([api_key, vrp_params[:name], checksum].compact.join('_'), d_params.to_json) if OptimizerWrapper.config[:dump][:vrp]
-
-              APIBase.services(api_key)[:params_limit].merge(OptimizerWrapper.access[api_key][:params_limit] || {}).each{ |key, value|
-                next if vrp_params[key].nil? || value.nil? || vrp_params[key].size <= value
-
-                error!({
-                  status: 'Exceeded params limit',
-                  message: "Exceeded #{key} limit authorized for your account: #{value}. Please contact support or sales to increase limits."
-                }, 400)
-              }
-
-              vrp = ::Models::Vrp.create(vrp_params)
-              if !vrp.valid? || vrp_params.nil? || vrp_params.keys.empty?
-                vrp.errors.add(:empty_file, message: 'JSON file is empty') if vrp_params.nil?
-                vrp.errors.add(:empty_vrp, message: 'VRP structure is empty') if vrp_params&.keys&.empty?
-                error!({ status: 'Model Validation Error', message: vrp.errors }, 400)
-              else
-                ret = OptimizerWrapper.wrapper_vrp(api_key, APIBase.services(api_key), vrp, checksum)
-                if ret.is_a?(String)
-                  # present result, with: VrpResult
-                  status 201
-                  present({ job: { id: ret, status: :queued }}, with: Grape::Presenters::Presenter)
-                elsif ret.is_a?(Hash)
-                  status 200
-                  if vrp.restitution_csv
-                    present(OptimizerWrapper.build_csv(ret.deep_stringify_keys), type: CSV)
-                  else
-                    present({ solutions: [ret], job: { status: :completed }}, with: Grape::Presenters::Presenter)
-                  end
+          post do # rubocop:disable Metrics/BlockLength
+            # Api key is not declared as part of the VRP and must be handled carefully and separatly from other parameters
+            api_key = params[:api_key]
+            checksum = Digest::MD5.hexdigest Marshal.dump(params)
+            d_params = declared(params, include_missing: false)
+            vrp_params = d_params[:points] ? d_params : d_params[:vrp]
+            APIBase.dump_vrp_dir.write([api_key, vrp_params[:name], checksum].compact.join('_'), vrp_params.to_json) if OptimizerWrapper.config[:dump][:vrp]
```

Comment:
There is still a modification on this line. Current code is as follows.
```suggestion
            APIBase.dump_vrp_dir.write([api_key, vrp_params[:name], checksum].compact.join('_'), d_params.to_json) if OptimizerWrapper.config[:dump][:vrp]
```
 If we want to be able to re-post a vrp launched and dumped via front "the same way" then we need to dump d_params, right ?